### DISCUSSION TITLE
ca65 .include

### DISCUSSION THREAD

**Discussion by AshtonSnapp** [cc65/cc65#2145] [Q&A]
Hello,

I am learning how to use the cc65 tool-chain for an upcoming project. Given the nature of the project, using the actual C compiler would probably very inefficient for e.g. an 8K system ROM. Therefore I will be using the ca65 assembler, and will be spliting the project over multiple files.

However, it is not entirely clear to me how combining those files together would even work. From the little I can tell, it is not as simple as just writing `.include "filename.a65"` or similar on a line and then the file's contents are included. Probably because that's `ld65`'s job. So how does this operate? The users guide example shows including a `.inc` file - does this file need any sort of special format in order to be valid? Do I need to create one for each file in my program, like one would for a C header file? It really doesn't include any information on including besides the macro/control command you type out.

**Comment by polluks** [comment]
Just take a look at the many projects so far
https://github.com/cc65/wiki/wiki/Applications-written-in-assembler-with-ca65
